### ISSUE TITLE
Should pick up panic!(format!(

### ISSUE THREAD

**Issue by heca-project** [rust-lang/rust-clippy#5013]
clippy should detect unnecessary `format!` blocks, such as `panic!(format!(` and others.

**Comment by blasrodri** [comment]
> clippy should detect unnecessary `format!` blocks, such as `panic!(format!(` and others.

Can you be more specific in which cases this should be added? I guess you refer to some sort of short circuiting that makes some expression to remain unevaluated no matter what. Is that correct? 

In particular, the case with `panic!` and `format!` does not follow those rules. For instance

```rust
#[derive(Debug)]
struct Whatever;

fn main() {
  let b = Whatever;
  panic!(format!("^^ {:?} ^^ ", b)) // prints ^^ Whatever ^^
}
```

**Comment by flip1995** [comment]
`panic!(format!(..))` is always weird, since `panic!()` uses the same formatting syntax as `fromat!` and you could just remove the `format!` and it'll have the same output.

**Comment by flip1995** [comment]
This should also warn on other macros of the panic family `unreachable`/`assert`/...

**Comment by camsteffen** [comment]
Added in #7743
